content/blogs/gui-setup-part1-setupapi.md

```diff
@@ -0,0 +1,104 @@
+
+During this first month (September 2023), my goal was to partly sync the code of the `setupapi.dll` module with Wine's.
+Indeed, this module was [forked](https://git.reactos.org/?p=reactos.git;a=blob;f=media/doc/WINESYNC.txt;h=f6614ac83a968cd5d7ddde6483dc85b94d264fc4;hb=HEAD#l182) way back in 2005, with later partial synchronizations of its code for some of its functions.
+
+For those of you who do not know what [setupapi](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/setupapi) is, it is a module that:
+- exposes [generic setup functions](https://learn.microsoft.com/en-us/windows/win32/setupapi/setup-api-portal) for reading and processing INF files, moving/copying files from an installation source media to a target..., supporting also extraction from compressed .CAB cabinet files;
+- and exposes device installation functions as well.
+
+Because of its very nature, it is "one of those Wine components we don't share completely because of architectural differences" [(CORE-7662)](https://jira.reactos.org/browse/CORE-7662) (due mainly to the device installation functions), and as such, as time goes by, Wine code improvements for code that is not tied to OS- and architecture-specific aspects might go unnoticed and our shared code slowly drifts apart for no good reason.
+
+As part of the 1st-stage GUI setup, I need to use the [File Queues](https://learn.microsoft.com/en-us/windows/win32/setupapi/file-queues) and [Cabinet Files](https://learn.microsoft.com/en-us/windows/win32/setupapi/cabinet-files--setup-api-) functions.
+These are those that have received recent improvements/bug-fixes from Wine, and that can be synchronized without too much trouble.
+(NOTE: It should be observed that our current text-mode 1st-stage installer, USETUP, re-implements a very small subset of these functions, used for extracting files from the `reactos.cab` source file and copying them to the ReactOS installation directory.
+In the GUI setup, my aim is to use directly those exposed by `setupapi.dll`.)
+
+The different sub-tasks I need to address are listed in this JIRA report [CORE-19209](https://jira.reactos.org/browse/CORE-19209).
+
+
+## Winesync.py script
+
+This endeavour started by trying to use our in-house `winesync.py` python script to help syncing Wine code.
+It requires one to write a configuration file (e.g. `setupapi.cfg`) that lists which files/directories can be synced for the given module, and maps these from their names in the [Wine repository](https://github.com/wine-mirror/wine/tree/master/), for example: [`dlls/setupapi`](https://github.com/wine-mirror/wine/tree/master/dlls/setupapi) (or one or more of its files), to their corresponding path names in the [ReactOS repository](https://github.com/reactos/reactos/tree/master/dll/win32/setupapi).
+
+As it is customary with any ReactOS developing exercise, one always encounters problems or bugs, and my attempt at using the `winesync.py` script was no exception to this rule.
```

Comment:
```suggestion
With any ReactOS development task, one always encounters problems or bugs. My attempt at using the `winesync.py` script was no exception to this rule.
```

This sentence has more then one comma and it's a little hard to understand, this is my interpretation